Kudu.Core/Deployment/StructuredTextLogger.cs

```diff
@@ -0,0 +1,120 @@
+﻿using Kudu.Core.Tracing;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text.RegularExpressions;
+
+namespace Kudu.Core.Deployment
+{
+    public class StructuredTextLogger : IDetailedLogger
+    {
+        private readonly IAnalytics _analytics;
+        private readonly int _depth;
+        private readonly string _path;
+        private const string LogEntryTemplate = "message=\"{0}\" time=\"{1}\" id=\"{2}\" type=\"{3}\"";
+        private const string LogEntryPattern = "message=\"(.*?)\"\\s+time=\"(.*?)\"\\s+id=\"(.*?)\"\\s+type=\"(.*?)\"";
+        private readonly Regex LogEntryRegex = new Regex(LogEntryPattern, RegexOptions.IgnoreCase | RegexOptions.Singleline);
+        private readonly StructuredTextDocument<LogEntry> _structuredTextDocment;
+
+        private static IEnumerable<KeyValuePair<string, string>> EscapeChars = new Dictionary<string, string>
```

Comment:
Shall we also define "EscapeChars" as a Dictionary? so that when we find char to escape/replace, we can do containsKey instead of loop thru the collection one by one?